src/models/tabularmodel.jl

```diff
@@ -0,0 +1,158 @@
+"""
+    emb_sz_rule(n_cat)
+
+Returns an embedding size corresponding to the number of classes for a 
+categorical variable using the rule of thumb present in python fastai.
+(see https://github.com/fastai/fastai/blob/2742fe844573d06e700f869839fb9ec5f3a9bca9/fastai/tabular/model.py#L12)
+"""
+emb_sz_rule(n_cat) = min(600, round(Int, 1.6 * n_cat^0.56))
+
+"""
+    get_emb_sz(cardinalities, [size_overrides])
+
+Returns a collection of tuples containing embedding dimensions corresponding to 
+number of classes in categorical columns present in `cardinalities` and adjusting for NaNs. 
+
+## Keyword arguments
+
+- `size_overrides`: A collection of Integers and `nothing` where the integer present at any index 
+    will be used to override the rule of thumb for getting embedding sizes.
+"""
+
+get_emb_sz(cardinalities::AbstractVector{<:Integer}, size_overrides=fill(nothing, length(cardinalities))) =
+    map(zip(cardinalities, size_overrides)) do (cardinality, override)
+        emb_dim = isnothing(override) ? emb_sz_rule(cardinality + 1) : Int64(override)
+        return (cardinality + 1, emb_dim)
+    end
+
+"""
+    get_emb_sz(cardinalities, categorical_cols, [size_overrides])
+
+Returns a collection of tuples containing embedding dimensions corresponding to 
+number of classes in categorical columns present in `cardinalities` and adjusting for NaNs. 
+
+## Keyword arguments
+
+- `size_overrides`: An indexable collection with column name as key and size
+    to override it with as the value.
+- `categorical_cols`: A collection of categorical column names.
+"""
+
+function get_emb_sz(cardinalities::AbstractVector{<:Integer}, categorical_cols::Tuple, size_overrides=Dict())
+    keylist = keys(size_overrides)
+    overrides = collect(map(categorical_cols) do col
+        col in keylist ? size_overrides[col] : nothing
+    end)
+    get_emb_sz(cardinalities, overrides)
+end
```

Comment:
I don't think being able to override with a dictionary is as useful when `cardinalities` is still a vector.

```suggestion
function get_emb_sz(cardinalities::Dict{<:Any, <:Integer}, size_overrides=Dict())
    values_and_overrides = map(pairs(cardinalities)) do (col, cardinality)
        cardinality, get(size_overrides, col, nothing)
    end
    get_emb_sz(first.(values_and_overrides), last.(values_and_overrides))
end
```